xeth/xeth.go

```diff
@@ -386,7 +386,7 @@ func (self *XEth) CurrentBlock() *types.Block {
 }
 
 func (self *XEth) GetBlockReceipts(bhash common.Hash) types.Receipts {
-	return self.backend.BlockProcessor().GetBlockReceipts(bhash)
+	return core.GetBlockReceipts(self.backend.ChainDb(), bhash)
```

Comment:
The problem with directly accessing the database is that you lose the caching done in the blockchain. I'd opt to call the `BlockChain` getter instead and let it cache the result for future use.